tasks/syncthing.yml

```diff
@@ -1,100 +1,149 @@
         owner={{syncthing_user}} group={{syncthing_user}} mode=700
 
 - name: create `bin` folder
+  become: true
   file: path={{syncthing_home}}/bin state=directory
         owner={{syncthing_user}} group={{syncthing_user}} mode=700
 
 - name: check syncthing version
+  become: true
+  become_user: "{{ syncthing_user }}"
   shell: "{{syncthing_home}}/bin/syncthing --version | awk '{ print $2 }' | cut -c 2-"
   changed_when: false
   ignore_errors: true
   register: version
 
+- debug: msg="version={{version}}, mine={{syncthing_version}}"
+
 - name: download
+  become: true
+  become_user: "{{ syncthing_user }}"
   get_url: url="{{syncthing_url}}" dest=/tmp/syncthing_{{syncthing_version}}.tar.gz
-  when: version|failed or version.stdout != "{{syncthing_version}}"
+  when: version|failed or version.stdout | version_compare("{{syncthing_version}}", '<')
 
 - name: extract
+  become: true
+  become_user: "{{ syncthing_user }}"
   command: tar xvzf /tmp/syncthing_{{syncthing_version}}.tar.gz -C /tmp
-  when: version|failed or version.stdout != "{{syncthing_version}}"
-
+  when: version|failed or version.stdout | version_compare("{{syncthing_version}}", '<')
 
 - name: copy executable to home dir
+  become: true
+  become_user: "{{ syncthing_user }}"
   shell: cp /tmp/syncthing-*{{syncthing_version}}/syncthing {{syncthing_home}}/bin/syncthing
-  when: version|failed or version.stdout != "{{syncthing_version}}"
+  when: version|failed or version.stdout | version_compare("{{syncthing_version}}", '<')
 
 - name: set ownership
+  become: true
   file: path={{syncthing_home}} recurse=yes owner={{syncthing_user}} group={{syncthing_user}}
 
 - name: supervisor | install
+  become: true
   apt: pkg=supervisor state=present
   when: syncthing_use_supervisor
 
 - name: supervisor | install config
+  become: true
   template: src=supervisor.conf dest=/etc/supervisor/conf.d/syncthing.conf mode=755
   register: supervisor_conf
   when: syncthing_use_supervisor
 
 - name: supervisor | reload
+  become: true
   command: supervisorctl reload
   when: syncthing_use_supervisor and supervisor_conf|changed
 
 - name: systemd | install service
+  become: true
   template: src=syncthing.service dest=/etc/systemd/system/syncthing.service mode=755
   register: systemd_conf
   when: syncthing_use_systemd
 
 - name: systemd | activate service
+  become: true
   service: name=syncthing enabled=yes state=started
   when: syncthing_use_systemd
 
 - name: systemd | reload service
-  service: name=syncthing state=reloaded
+  become: true
+  service: name=syncthing state=restarted
   when: syncthing_use_systemd and systemd_conf|changed
 
+- name: upstart | install service
+  become: true
+  template: src=upstart.conf dest=/etc/init/syncthing.conf mode=755
+  register: upstart_conf
+  when: syncthing_use_upstart
+
+- name: upstart | activate service
+  become: true
+  service: name=syncthing enabled=yes state=started
+  when: syncthing_use_upstart
```

Comment:
This could be reduced to one statement. Something like:
```
- name: activate service
  become: true
  service: name=syncthing enabled=yes state=started
  when: syncthing_use_upstart or syncthing_use_systemd
```